tests/tas-elasticsearch/src/test/java/org/alfresco/elasticsearch/AlfrescoStackInitializer.java

```diff
@@ -0,0 +1,139 @@
+package org.alfresco.elasticsearch;
+
+import org.springframework.context.ApplicationContextInitializer;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.test.context.support.TestPropertySourceUtils;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.Network;
+import org.testcontainers.containers.PostgreSQLContainer;
+import org.testcontainers.containers.wait.strategy.LogMessageWaitStrategy;
+import org.testcontainers.containers.wait.strategy.Wait;
+import org.testcontainers.elasticsearch.ElasticsearchContainer;
+import org.testcontainers.lifecycle.Startables;
+
+import java.time.Duration;
+import java.util.concurrent.CompletableFuture;
+
+public class AlfrescoStackInitializer implements ApplicationContextInitializer<ConfigurableApplicationContext>
+{
+
+    public static Network network;
+
+    public static GenericContainer alfresco;
+
+    public static ElasticsearchContainer elasticsearch;
+
+    public static GenericContainer liveIndexer;
+
+    private static String ES_CONNECTOR_TAG = "3.0.0-RC1";
+
+    @Override
+    public void initialize(ConfigurableApplicationContext configurableApplicationContext)
+    {
+
+        network = Network.newNetwork();
+
+        alfresco = new GenericContainer("quay.io/alfresco/alfresco-content-repository:latest")
+                           .withEnv("JAVA_TOOL_OPTIONS",
+                                    "-Dencryption.keystore.type=JCEKS " +
+                                    "-Dencryption.cipherAlgorithm=DESede/CBC/PKCS5Padding " +
+                                    "-Dencryption.keyAlgorithm=DESede " +
+                                    "-Dencryption.keystore.location=/usr/local/tomcat/shared/classes/alfresco/extension/keystore/keystore " +
+                                    "-Dmetadata-keystore.password=mp6yc0UD9e -Dmetadata-keystore.aliases=metadata " +
+                                    "-Dmetadata-keystore.metadata.password=oKIWzVdEdA -Dmetadata-keystore.metadata.algorithm=DESede")
+                           .withEnv("JAVA_OPTS",
+                                    "-Delasticsearch.createIndexIfNotExists=true " +
+                                    "-Ddb.driver=org.postgresql.Driver " +
+                                    "-Ddb.username=alfresco " +
+                                    "-Ddb.password=alfresco " +
+                                    "-Ddb.url=jdbc:postgresql://postgres:5432/alfresco " +
```

Comment:
We could extract port numbers for all the services, but I don't think there's actually any point.  Hard-coding these looks good to me.

Reply:
not needed: the ports are user "internally" , inside the docker network. If you need to interact with a container you need the mapped port to the host, e,g: `container.getMappedPort(8080)`